platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/SamlValidator.java

```diff
@@ -0,0 +1,415 @@
+import org.opensaml.saml2.core.LogoutResponse;
+import org.opensaml.xml.XMLObject;
+import org.opensaml.xml.validation.ValidatingXMLObject;
+import org.opensaml.xml.validation.ValidationException;
+import org.opensaml.xml.validation.Validator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.samlp.SamlProtocol;
+import ddf.security.samlp.SimpleSign;
+
+public abstract class SamlValidator implements Validator<ValidatingXMLObject> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SamlValidator.class);
+
+    protected final Builder builder;
+
+    private SamlValidator(Builder builder) {
+        this.builder = builder;
+    }
+
+    @Override
+    public final void validate(ValidatingXMLObject xmlObject) throws ValidationException {
+        // This is intentionally an instance equality check
+        if (xmlObject != builder.xmlObject) {
+            throw new ValidationException("Cannot validate a different target.");
+        }
+
+        checkTimestamp();
+        checkSamlVersion();
+        checkId();
+        checkRequiredFields();
+        checkDestination();
+        additionalValidation();
+    }
+
+    protected void checkTimestamp() throws ValidationException {
+        DateTime issueInstant = getIssueInstant();
+        if (issueInstant == null) {
+            throw new ValidationException("Issue Instant cannot be null!");
+        }
+
+        Instant instant = Instant.ofEpochMilli(issueInstant.getMillis());
+        Instant now = Instant.now();
+        if (instant.minus(builder.jitter)
+                .isAfter(now)) {
+            throw new ValidationException("Issue Instant cannot be in the future");
+        }
+
+        if (instant.plus(builder.jitter)
+                .isBefore(now.minus(builder.issueTimeout))) {
+            throw new ValidationException("Issue Instant was outside valid time range");
+        }
+
+    }
+
+    protected void checkSamlVersion() throws ValidationException {
+        SAMLVersion samlVersion = getSamlVersion();
+        if (samlVersion == null) {
+            throw new ValidationException("SAML Version cannot be null!");
+        }
+        if (!SAMLVersion.VERSION_20.equals(samlVersion)) {
+            throw new ValidationException("Invalid SAML Version!");
+        }
+    }
+
+    protected abstract void checkRequiredFields() throws ValidationException;
+
+    protected abstract void checkDestination() throws ValidationException;
+
+    protected abstract void additionalValidation() throws ValidationException;
+
+    protected void checkId() throws ValidationException {
+        // pass, default method
+    }
+
+    protected abstract DateTime getIssueInstant();
+
+    protected abstract SAMLVersion getSamlVersion();
+
+    protected void checkPostSignature() throws ValidationException {
+        // noop, default method
+    }
+
+    void checkPostSignature(SignableSAMLObject samlObject) throws ValidationException {
```

Comment:
This validator isn't going to work for the SSO HoK spec. We might want to take a look at that as well so that this can be ready for when we support that spec.